internal/context/workspace/domain/workflow/reader.go

```diff
@@ -200,4 +212,172 @@ func parseWorkflowParamValue(value string) (paramType string, optional bool, def
 
 // NextflowReader reader for workflow written by Nextflow
 type NextflowReader struct {
+	inputParams   []WorkflowParam
+	outputParams  []WorkflowParam
+	graphFilepath string
+}
+
+// ValidateWorkflowFiles ...
+func (r *NextflowReader) ValidateWorkflowFiles(ctx context.Context, version *WorkflowVersion, baseDir, mainWorkflowPath string) error {
+	// check whether the main workflow file exists
+	if mainWorkflowPath != "main.nf" {
+		applog.Errorw("wrong name", "name", mainWorkflowPath)
+		return apperrors.NewInternalError(fmt.Errorf("the name of main workflow file is wrong"))
+	}
+	if !utils.FileExists(path.Join(baseDir, mainWorkflowPath)) {
+		applog.Errorw("the main workflow file not found", "filepath", path.Join(baseDir, mainWorkflowPath))
+		return apperrors.NewInternalError(fmt.Errorf("the main workflow file not found"))
```

Comment:
Maybe `NewNotFoundError`?

Reply:
There are so many errors thrown in the project that it doesn't do much to declare a variable for each error.